skypy/halo/abundance_matching.py

```diff
@@ -44,6 +46,13 @@ def vale_ostriker(halos, galaxies, mass='mass', luminosity='luminosity',
     .. [1] Vale A., Ostriker J. P., 2004, MNRAS, 353, 189
     """
 
+    if subhalos:
+        if mass not in halos.columns:
+            raise ValueError("{} is not a column in halos".format(mass))
+        if mass not in subhalos.columns:
+            raise ValueError("{} is not a column in subhalos".format(mass))
+        halos = vstack((halos, subhalos))
+
```

Comment:
What if my halo mass column is `halo_mass` and my subhalo mass column is `subhalo_mass`? Wouldn't it be much easier to pass columns/arrays instead of tables?

Reply:
Yes perhaps. The restriction then would be that you couldn't trivially "carry over" any other existing columns in the (sub)halo / galaxy tables into the new matched table.

Reply:
In that case, wouldn't it make more sense to have the user call `astropy.table.vstack` explicitly on the halos and subhalos tables, since they need to have matching definitions for the stacking to work?

Reply:
For the user to have to call vstack as part of their pipeline is perhaps a bit unintuitive? I like the idea of wrapping it in an abundance matching function in skypy, although perhaps this PR isn't the best design. I think you mentioned before the idea of incorporating the entire Vale & Ostriker 2004 model into a single function, which would be the other extreme.

Reply:
Do we need to keep track of which subhalos correspond to a parent halo here?